keys/keys.go

```diff
@@ -369,3 +371,251 @@ func Range(ba roachpb.BatchRequest) roachpb.RSpan {
+		return buf.String()
+	}
+
+	// get range id
+	var i int64
+	var err error
+	key, i, err = encoding.DecodeVarint(key)
+	if err != nil {
+		fmt.Fprintf(&buf, "/err<%v:%s>", err, key.String())
+		return buf.String()
+	}
+
+	fmt.Fprintf(&buf, "/%s", parser.DInt(i))
+
+	// get suffix
+	suffixDict := []struct {
+		name   string
+		suffix []byte
+		ppFunc func(key roachpb.Key) string
+	}{
+		{name: "ResponseCache", suffix: LocalResponseCacheSuffix, ppFunc: responseCacheKeyPrint},
+		{name: "RaftLeaderLease", suffix: localRaftLeaderLeaseSuffix},
+		{name: "RaftTombstone", suffix: localRaftTombstoneSuffix},
+		{name: "RaftHardState", suffix: localRaftHardStateSuffix},
+		{name: "RaftAppliedIndex", suffix: localRaftAppliedIndexSuffix},
+		{name: "RaftLog", suffix: localRaftLogSuffix, ppFunc: raftLogKeyPrint},
+		{name: "RaftTruncatedState", suffix: localRaftTruncatedStateSuffix},
+		{name: "RaftLastIndex", suffix: localRaftLastIndexSuffix},
+		{name: "RangeGCMetadata", suffix: localRangeGCMetadataSuffix},
+		{name: "RangeLastVerificationTimestamp", suffix: localRangeLastVerificationTimestampSuffix},
+		{name: "RangeStats", suffix: localRangeStatsSuffix},
+	}
+	hasSuffix := false
+	for _, s := range suffixDict {
+		if bytes.HasPrefix(key, s.suffix) {
+			fmt.Fprintf(&buf, "/%s", s.name)
+			key = key[len(s.suffix):]
+			if s.ppFunc != nil && len(key) != 0 {
+				fmt.Fprintf(&buf, "%s", s.ppFunc(key))
+				return buf.String()
+			}
+			hasSuffix = true
+			break
+		}
+	}
+
+	// get encode values
+	if hasSuffix {
+		fmt.Fprintf(&buf, "%s", decodeKeyPrint(key))
+	} else {
+		fmt.Fprintf(&buf, "%s", key.String())
+	}
+
+	return buf.String()
+}
+
+func localRangeKeyPrint(key roachpb.Key) string {
+	var buf bytes.Buffer
+
+	suffixDict := []struct {
+		name   string
+		suffix []byte
+		atEnd  bool
+	}{
+		{name: "RangeDescriptor", suffix: LocalRangeDescriptorSuffix, atEnd: true},
+		{name: "RangeTreeNode", suffix: localRangeTreeNodeSuffix, atEnd: true},
+		{name: "Transaction", suffix: localTransactionSuffix, atEnd: false},
+	}
+
+	for _, s := range suffixDict {
+		if s.atEnd {
+			if bytes.HasSuffix(key, s.suffix) {
+				key = key[:len(key)-len(s.suffix)]
+				fmt.Fprintf(&buf, "/%s%s", s.name, decodeKeyPrint(key))
+				return buf.String()
+			}
+		} else {
+			begin := bytes.Index(key, s.suffix)
+			if begin > 0 {
+				addrKey := key[:begin]
+				id := key[(begin + len(s.suffix)):]
+				fmt.Fprintf(&buf, "/%s/addrKey:%s/id:%s", s.name, decodeKeyPrint(addrKey), id.String())
+				return buf.String()
+			}
+		}
+	}
+	fmt.Fprintf(&buf, "%s", decodeKeyPrint(key))
+	return buf.String()
+}
+
+func hexPrint(key roachpb.Key) string {
+	return fmt.Sprintf("/%s", key.String())
+}
+
+// TableDataPrefix
+func decodeKeyPrint(key roachpb.Key) string {
+	var buf bytes.Buffer
+	for k := 0; len(key) > 0; k++ {
+		var d interface{}
+		var err error
+		switch encoding.PeekType(key) {
+		case encoding.Null:
+			key, _ = encoding.DecodeIfNull(key)
+			d = parser.DNull
```

Comment:
Rather than initializing `d`, you could avoid the `parser` dependency by outputting the value:

```
  fmt.Fprintf(&buf, "/NULL")
```